### PR TITLE
chore: add code path handling and logs

### PR DESCRIPTION
**Pull request by srijan-deepsource**

### CODE REVIEW

File: run_community_analyzer.py [hunk 1]
```diff
@@ -9,8 +10,17 @@
 
 sentry.initialize()
 
+logger = logging.getLogger("run_community_analyzer")
```
Comment (eshaan-deepsource) [inline run_community_analyzer.py]: can't we just use `__name__`?

File: sarif-parser/src/sarif_parser/__init__.py [hunk 2]
```diff
@@ -3,11 +3,19 @@
 
 import hashlib
 import json
+import logging
 import os.path
 from typing import Any, Sequence, TypedDict, Union
 
 import sentry
 
+logger = logging.getLogger("sarif_parser")
```
Comment (eshaan-deepsource) [inline sarif-parser/src/sarif_parser/__init__.py]: same as above

File: run_community_analyzer.py (hunk 1 above)
Comment (srijan-deepsource) [inline run_community_analyzer.py]:
```suggestion
logger = logging.getLogger(__name__)
```

File: sarif-parser/src/sarif_parser/__init__.py (hunk 2 above)
Comment (srijan-deepsource) [inline sarif-parser/src/sarif_parser/__init__.py]:
```suggestion
logger = logging.getLogger(__name__)
```

**Review by tushar-deepsource — APPROVED** [review]

**Pull request merged**